### ISSUE TITLE
Tabs on bottom messed up in Derveloper 133.0b1

### ISSUE THREAD

**Issue by plissken1138** [MrOtherGuy/firefox-csshacks#438]
Tabs on bottom messed up in Developer 133.0b1

It was working fine up until latest update. In addition to that,, not to make only one thing annoying, 
clicking into address bar now opens it to the whole width of the screen...

**Comment by plissken1138** [comment]
looks like tabs_on_bottom_v2 still works.

address bar fixed when I removed megabar-tweaks.css

**Comment by MrOtherGuy** [comment]
Yeah, there's a couple of these "v2" styles that will replace the old ones in 133 onwards. I want to preserve the old ones though for the lifetime of the ESR 128 cycle so for now there are two versions.

**Comment by madeddy** [comment]
_Not sure if this fits here or should i make a new ish?_

The "tabs_on_bottom_v2.css" does work for me in the v133 _release_ and I'm in _Kubuntu 22.04.5_. I use it as _import_ and have the two other snippeds i have even outcommented for test purpose.

```
/*@import url(window_control_placeholder_support.css);
@import url(linux_gtk_window_control_patch.css);*/
@import url(/chrome/tabs_on_bottom_v2.css);

/* Your other/personal rules here */
```
_toolkit.legacyUserProfileCustomizations.stylesheets_ is also set to **true**

**Comment by artemlive** [comment]
It's the same to me. 
133.0 (aarch64)

**Comment by CurePindal** [comment]
tabs_on_bottom_v2 messes up the buttons for me:
![FF133](https://github.com/user-attachments/assets/2f867546-61bd-4573-8a94-5ec850456f28)
Thats whats called "buttonbox" thats misaligned, right?

Anybody an idea whats wrong? 
I only use:
@import url(window_control_placeholder_support.css);
@import url(tabs_on_bottom_v2.css);

**Comment by madeddy** [comment]
Update: Tabs on bottom works now for me with
```
import url(window_control_placeholder_support.css);
import url(tabs_on_bottom_v2.css);
```
but i have now also the additional space on my addressbar(as @CurePindal).

**Comment by MrOtherGuy** [comment]
Hey dudes, do not use window_control_placeholder_support.css with tabs_on_bottom_v2 - you do not need it. Whatever it may cause is completely unintentional.

**Comment by artemlive** [comment]
What am I doing wrong? 
userChrome.css
`import url(chrome/tabs_on_bottom_v2.css); `

> toolkit.legacyUserProfileCustomizations.stylesheets	true

**Comment by madeddy** [comment]
@CurePindal This strange "space" in the addressbar is only from use of _window_control_placeholder_support.css_ and IMO not related. On my sys this happens also if i use this **without** _tabs_on_bottom_v2.css_.

**Comment by MrOtherGuy** [comment]
> What am I doing wrong? userChrome.css: `import url(chrome/tabs_on_bottom_v2.css); `
> 
> > toolkit.legacyUserProfileCustomizations.stylesheets	true
> 
> and it doesn't work 133.0 (aarch64)

Import syntax is `@import url` so try that. See https://developer.mozilla.org/en-US/docs/Web/CSS/@import

**Comment by artemlive** [comment]
@MrOtherGuy ty for the reply. 
It doesn't work. 

```
pwd
/Users/myuser/Library/Application Support/Firefox/Profiles/lezmnfkg.default-release

cat chrome/userChrome.css --style=plain                                                                                                                              
@import url(chrome/tabs_on_bottom_v2.css);

ls chrome/chrome/tabs_on_bottom_v2.css                                                                                                                            
chrome/chrome/tabs_on_bottom_v2.css
```

**Comment by MrOtherGuy** [comment]
> I don't want to post the whole file, so (I run it within the root profile's directory, so there is the "chrome" directory twice):

Note that all `@import` statements must be placed before any other rules in the file otherwise the import is ignored.

**Comment by artemlive** [comment]
> > I don't want to post the whole file, so (I run it within the root profile's directory, so there is the "chrome" directory twice):
> 
> Note that all `@import` statements must be placed before any other rules in the file otherwise the import is ignored.
I get that, thanks. My **userChrome.css** file has only one line: `@import url(chrome/tabs_on_bottom_v2.css);`

**Comment by artemlive** [comment]
Hmm, I see that in the default style, it looks like this: 
<img width="841" alt="scr 2024-11-27 at 15 27 17" src="https://github.com/user-attachments/assets/26d5e0d0-0c34-42c9-909c-43ad5c7c8c7d">

But If I include the tabs_on_bottom_v2.css:
<img width="799" alt="scr 2024-11-27 at 15 28 15" src="https://github.com/user-attachments/assets/eeab2553-d9ea-4502-85c8-c5b58ec653f6">

So, it moves under the address bar but not to the bottom of the window.

**Comment by MrOtherGuy** [comment]
Right, well tabs_on_bottom.css (and tabs_on_bottom_v2.css by extension) is **supposed** to move tabs as the bottom-most *toolbar*. So I take it that it does work? 

If you want tabs below content, then use [tabs_below_content_v2.css](https://github.com/MrOtherGuy/firefox-csshacks/blob/master/chrome/tabs_below_content_v2.css)

@CurePindal can you confirm that tabs_on_bottom_v2.css works for you as well **after you remove anything that has to do with old tabs_on_bottom.css and window_control_placeholder_support.css?**

**Comment by artemlive** [comment]
> Right, well tabs_on_bottom.css (and tabs_on_bottom_v2.css by extension) is **supposed** to move tabs as the bottom-most _toolbar_. So I take it that it does work?
> 
> If you want tabs below content, then use [tabs_below_content_v2.css](https://github.com/MrOtherGuy/firefox-csshacks/blob/master/chrome/tabs_below_content_v2.css)
> 
> @CurePindal can you confirm that tabs_on_bottom_v2.css works for you as well **after you remove anything that has to do with old tabs_on_bottom.css and window_control_placeholder_support.css?**

The problem was that I misinterpreted the `tabs_on_bottom_v2.css` as a replacement for the `tabs_on_bottom.css` that I used for moving the tabs to the bottom of the window (below content) before the latest upgrade.  
I can confirm that everything works as expected. Thanks, and sorry for the mess.

**Comment by MrOtherGuy** [comment]
No worries, although I don't know what you had done previously, but tabs_on_bottom.css never moved tabs below content. Perhaps you had code from old tabs_below_cotent.css but labelled it as if it was tabs_on_bottom.

Who knows, main thing is that things works fine now.

**Comment by Laephis** [comment]
Starting to pull my hair out over this, but I've read through this thread multiple times and I still can't get tabs_on_bottom_v2.css working no matter what I do.  I have my userChrome.css file looking like this:

`import url(tabs_on_bottom_v2.css);`

and `toolkit.legacyUserProfileCustomizations.stylesheets ` set to true but no effect.  I'm just updating a previously working tweak, which I've been doing for years now since FF devs refuse to give us this option built-in. 

Am I just missing something painfully obvious?

**Comment by MrOtherGuy** [comment]
@Laephis you could try just copying the contents of tabs_on_bottom_v2.css directly into userChrome.css to rule out any issues from potentially incorrect use of `@import`. For good measure, also remove everything else in your userChrome.css file - just copy things temporarily to somewhere else.

**Comment by MrOtherGuy** [comment]
Please keep this issue about tabs_on_bottom.css and open new issues about unrelated things.

**Comment by MrOtherGuy** [comment]
Okay, glad you sorted that out. But in addition to issues that are simply about loading the file such was the case here, you should know that there is also quite a lot of platform dependent styles and some other differences in Firefox so there is very much a possibility for styles not behaving the same across systems.

Moreover, various LInux distributions are known to add their own changes into Firefox and pushing the modified version to their package manager. Those changes could be anything and I cannot really account for whatever random stuff they might do - thus I would recommend that if you are customizing Firefox then you should get your build directly from Mozilla instead of your package manager.
